FirebaseStorageSwift/Tests/Integration/StorageIntegration.swift

```diff
@@ -0,0 +1,598 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseCore
+import FirebaseStorage
+import FirebaseStorageSwift
+import XCTest
+
+class StorageIntegration: XCTestCase {
+  var app: FirebaseApp!
+  var storage: Storage!
+  static var once = false
+
+  override class func setUp() {
+    FirebaseApp.configure()
+  }
+
+  override func setUp() {
+    super.setUp()
+    app = FirebaseApp.app()
+    storage = Storage.storage(app: app!)
+
+    if !StorageIntegration.once {
+      StorageIntegration.once = true
+      let setupExpectation = expectation(description: "setUp")
+
+      let largeFiles = ["ios/public/1mb"]
+      let emptyFiles =
+        ["ios/public/empty", "ios/public/list/a", "ios/public/list/b", "ios/public/list/prefix/c"]
+      setupExpectation.expectedFulfillmentCount = largeFiles.count + emptyFiles.count
+
+      do {
+        let bundle = Bundle(for: StorageIntegration.self)
+        let filePath = try XCTUnwrap(bundle.path(forResource: "1mb", ofType: "dat"),
+                                     "Failed to get filePath")
+        let data = try XCTUnwrap(try Data(contentsOf: URL(fileURLWithPath: filePath)),
+                                 "Failed to load file")
+
+        for largeFile in largeFiles {
+          let ref = storage.reference().child(largeFile)
+          _ = ref.putData(uploadData: data, metadata: nil, completion: { result in
+            switch result {
+            case let .success(metadata):
+              XCTAssertEqual(metadata.name, "1mb")
+              setupExpectation.fulfill()
+            case let .failure(error):
+              XCTFail("Unexpected error \(error) from putData")
+            }
+          })
+        }
+        for emptyFile in emptyFiles {
+          let ref = storage.reference().child(emptyFile)
+          _ = ref.putData(uploadData: data, metadata: nil, completion: { result in
+            switch result {
+            case let .success(metadata):
+              XCTAssertNotNil(metadata, "Metadata should not be nil")
+              setupExpectation.fulfill()
+            case let .failure(error):
+              XCTFail("Unexpected error \(error) from putData")
+            }
+          })
+        }
+      } catch {
+        XCTFail("Exception thrown setting up files in setUp")
+      }
+      waitForExpectations()
```

Comment:
This is going to wait for expectations even after an error is thrown. This should probably be inside the `do`/`catch` block.

Reply:
Done.